Sources/Overload/OvEditor/src/OvEditor/Panels/MenuBar.cpp

```diff
@@ -202,31 +202,30 @@ void OvEditor::Panels::MenuBar::CreateLayoutMenu()
 {
 	auto& layoutMenuList = CreateWidget<MenuList>("Layout");
 
+	auto& uiManager = *EDITOR_CONTEXT(uiManager);
+	std::string layoutsPath = uiManager.GetLayoutsPath();
+
 	auto& saveMenuItem = layoutMenuList.CreateWidget<MenuItem>("Save");
 	saveMenuItem.ClickedEvent += EDITOR_BIND(SaveCurrentLayout);
 
 	auto& saveNewMenuList = layoutMenuList.CreateWidget<MenuList>("Save New");
 	auto& layoutInputText = saveNewMenuList.CreateWidget<InputFields::InputText>("Layout Name");
 	layoutInputText.selectAllOnClick = true;
-	layoutInputText.EnterPressedEvent += [this](std::string p_input)
+	layoutInputText.EnterPressedEvent += [this, layoutsPath](std::string p_input)
 	{
 		if (p_input.empty())
 			return;
 
-		auto& UIManager = OvCore::Global::ServiceLocator::Get<OvUI::Core::UIManager>();
-		std::string layoutsPath = UIManager.GetLayoutsPath();
-		EDITOR_EXEC(DelayAction(std::bind(&OvUI::Core::UIManager::SaveLayout, &UIManager, layoutsPath + p_input + ".ini"), 1));
+		auto& uiManager = *EDITOR_CONTEXT(uiManager);
+		EDITOR_EXEC(DelayAction(std::bind(&OvUI::Core::UIManager::SaveLayout, &uiManager, layoutsPath + p_input + ".ini"), 1));
 	};
 
 	auto& loadMenuList = layoutMenuList.CreateWidget<MenuList>("Load");
 
-	loadMenuList.ClickedEvent += [&]
+	loadMenuList.ClickedEvent += [&, layoutsPath]
```

Comment:
You shouldn't need to copy `layoutsPath` if you already capture it by reference with `&`. I'd advise against using `&`, as it makes it hard to catch potential issues, so explicitly using `&layoutsPath` could be a good option. Not sure how many other references you need here